app/assets/javascripts/single_page/dynamic_table.js.erb

```diff
@@ -98,16 +104,29 @@ const handleCheck = (e) => (e.parents("table").DataTable().row(e.closest("tr")).
         });
       }
       if (options.assayId) this.assayId = options.assayId;
-      this.initHeader(columns);
       this.options = options;
+			this.initHeader(columns);
     },
     initHeader: function (columns) {
       const requiredCols = columns.map((c, i) => c.required && i).filter((c) => c);
+			if (this.options.readonly){
+				const sampleTypes = this.getSampleTypes()
+				$j.each(sampleTypes, (i, s) => {
+					if (!isNaN(s)) {
+						const colIdxs = this.table.columns(`${s}:name`)[0]
+						if (colIdxs.length) colIdxs.push(colIdxs[0]-1)
+						this.table.column(function ( idx, data, node ) {
+								if (colIdxs.includes(idx)) $j(node).addClass(`sp-variant-${i % 2}`)
+						})
+					}
+				})
+			}
       this.table
         .columns()
         .header()
         .each((x, i) => {
-          if (i && requiredCols.includes(i)) {
+					if (i == 0) $j(x).append(`<input type='checkbox' onClick='selectAll(this)'/>`)
```

Comment:
Indentation needs converting to spaces

Reply:
Although it's difficult to handle the formatting of `js.erb` files in VSCode, the indentation issue is fixed in the new commit.